pywebdriver/plugins/scale_protocols/mettler_toledo_8217.py

```diff
@@ -45,32 +45,7 @@ def _baudrate(self):
 
     def acquire_data(self, connection):
         """Acquire data over the connection."""
-        buffer = b""
-        stx = False
-        # ask for weight data
-        try:
-            connection.write(b"W")
-        except serial.SerialException as e:
-            raise ScaleConnectionError() from e
-        while True:
-            try:
-                c = connection.read(1)
-            except serial.SerialException as e:
-                raise ScaleConnectionError() from e
-            if not c:
-                # timeout
-                raise ScaleAcquireDataError("read time-out")
-            if c == b"\x02":
-                # start of answer
-                stx = True
-                buffer = b""
-            elif c == b"\r":
-                # end of answer
-                if not stx:
-                    continue
-                break
-            else:
-                buffer += c
+        buffer, stx = self._read_raw_data(connection)
```

Comment:
```suggestion
        buffer = self._read_raw_data(connection)
```
no need to return `stx`, it’s only used by the read loop.